nsxt/resource_nsxt_policy_context_profile_test.go

```diff
@@ -0,0 +1,346 @@
+					resource.TestCheckResourceAttr(testResourceName, "attribute.2585605962.sub_attribute.2040405228.value.3398512106", "CIFS_SMB_V1"),
+					resource.TestCheckResourceAttr(testResourceName, "attribute.2585605962.sub_attribute.2040405228.value.3787226665", "CIFS_SMB_V2"),
+					resource.TestCheckResourceAttr(testResourceName, "attribute.2585605962.value.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "attribute.2585605962.value.1810355442", "CIFS"),
+
+					resource.TestCheckResourceAttr(testResourceName, "attribute.951130521.data_type", "STRING"),
+					resource.TestCheckResourceAttr(testResourceName, "attribute.951130521.key", "URL_CATEGORY"),
+					resource.TestCheckResourceAttrSet(testResourceName, "attribute.951130521.is_alg_type"),
+					resource.TestCheckResourceAttr(testResourceName, "attribute.951130521.value.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "attribute.951130521.value.3317229948", "Abortion"),
+				),
+			},
+		},
+	})
+}
+
+func testAccNsxtPolicyContextProfileExists(resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy ContextProfile resource %s not found in resources", resourceName)
+		}
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy ContextProfile resource ID not set in resources")
+		}
+
+		err := nsxtPolicyContextProfileExists(resourceID)
+
+		if err != nil {
+			return fmt.Errorf("Error while retrieving policy ContextProfile ID %s. Error: %v", resourceID, err)
+		}
+
+		return nil
+	}
+}
+
+func nsxtPolicyContextProfileExists(resourceID string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	var err error
+	if testAccIsGlobalManager() {
+		nsxClient := gm_infra.NewDefaultContextProfilesClient(connector)
+		_, err = nsxClient.Get(resourceID)
+	} else {
+		nsxClient := infra.NewDefaultContextProfilesClient(connector)
+		_, err = nsxClient.Get(resourceID)
+	}
+	return err
+}
+
+func testAccNsxtPolicyContextProfileCheckDestroy(state *terraform.State, displayName string) error {
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_policy_context_profile" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		err := nsxtPolicyContextProfileExists(resourceID)
+		if err == nil {
+			return fmt.Errorf("Policy ContextProfile %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtPolicyContextProfileTemplate(name string, attributes string) string {
+	return fmt.Sprintf(`
+resource "nsxt_policy_context_profile" "test" {
+  display_name = "%s"
+  description  = "Acceptance Test"
+  tag {
+    scope = "color"
+    tag   = "orange"
+  }
+%s
+}`, name, attributes)
+}
+
+func testAccNsxtPolicyContextProfileAttributeDomainNameTemplate() string {
+	return fmt.Sprintf(`
+attribute {
+  data_type = "STRING"
```

Comment:
Please fix the spacing and indentations in the tests configurations here

Reply:
Sure. fixed